pennylane/templates/layers/gate_fabric.py

```diff
@@ -0,0 +1,211 @@
+        :target: javascript:void(0);
+
+    |
+
+    The individual four qubit :class:`~.DoubleExcitation()` and :class:`~.OrbitalRotation()` gates given here are equivalent to the
+    :math:`\text{QNP}_{PX}(\theta)` and :math:`\text{QNP}_{OR}(\phi)` gates presented in `arXiv:2104.05692 <https://arxiv.org/abs/2104.05695>`_
+    respectively. Moreover, regardless of the choice of :math:`\hat{\Pi}`, this gate fabric will exactly preserve the quantum numbers
+    :math:`\hat{N}_{\alpha}`, :math:`\hat{N}_{\beta}` and :math:`\hat{S}^{2}`.
+
+    Args:
+        weights (tensor_like): Array of weights of shape ``(D, L, 2)``\,
+            where ``D`` is the number of gate fabric layers and ``L = N/2-1``
+            is the number of :math:`\hat{Q}(\theta, \phi)` gates per layer.
+        wires (Iterable): wires that the template acts on
+        init_state (tensor_like): iterable of shape ``(len(wires),)``\, representing the Hartree-Fock state in Jordan-Wigner basis
+            that is used to initialize the wires.
+        include_pi (boolean): If ``include_pi = True``\, the optional constant :math:`\hat{\Pi}` gate  is set to :math:`\text{OrbitalRotation}(\pi)`.
+            Default value is :math:`\hat{I}`.
+
+    .. UsageDetails::
+
+        #. The number of wires :math:`N` has to be equal to the number of
+           spin orbitals included in the active space, and should be even.
+
+        #. The number of trainable parameters scales linearly with the number of layers as
+           :math:`2 D (N/2-1)`.
+
+        An example of how to use this template is shown below:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import GateFabric
+            from functools import partial
+
+            # Build the electronic Hamiltonian from a local .xyz file
+            h, qubits = qml.qchem.molecular_hamiltonian("h2", "h2.xyz")
+
+            # Define the Hartree-Fock state
+            electrons = 2
+            ref_state = qml.qchem.hf_state(electrons, qubits)
+
+            # Define the device
+            dev = qml.device('default.qubit', wires=qubits)
+
+            # Define the ansatz
+            ansatz = partial(GateFabric, init_state=ref_state, include_pi=True)
+
+            # Define the cost function
+            cost_fn = qml.ExpvalCost(ansatz, h, dev)
+
+            # Compute the expectation value of 'h'
+            layers = 2
+            params = qml.init.gate_fabric_normal(layers, qubits)
+            print(cost_fn(params))
+
+        **Parameter shape**
+
+        The shape of the weights argument can be computed by the static method
+        :meth:`~.GateFabric.shape` and used when creating randomly
+        initialised weight tensors:
+
+        .. code-block:: python
+
+            shape = GateFabric.shape(n_layers=2, n_wires=4)
+            weights = np.random.random(size=shape)
+
+
+    """
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
```

Comment:
@obliviateandsurrender is there a known gradient recipe for this gate? If not, best to have `grad_method=None`

Reply:
I have added the `grad_method=None`. I am assuming that by `gate` you meant the overall unitary block made by the template and not the gates (`OrbitalRotation` and `DoubleExcitation`) it is composed of because for them four-term parameter-shift rule exists.

Reply:
Also, would you have an idea about why in `Tests/core-tests(3.8, tf, torch, jax)`, the tests `TestPassthruIntegration::test_prob_differentiability` and `TestPassthruIntegration::test_backprop_gradient` are failing? I am unable to replicate the error in my local machine.

Reply:
> I have added the grad_method=None. I am assuming that by gate you meant the overall unitary block made by the template and not the gates (OrbitalRotation and DoubleExcitation) it is composed of because for them four-term parameter-shift rule exists.

Yep, exactly 👍 

>  I am unable to replicate the error in my local machine.

Oh this is very interesting, the same thing was happening to me in my PR yesterday: #1685. It must be related either to:

- A new release of JAX?
- A change in `master` that has affected the test suite

Reply:
There seems to have been a new release of JAX made on the 24th of September: https://pypi.org/project/jax/#history

Perhaps this is the cause

Reply:
Can confirm that this is it after checking locally. :slightly_smiling_face: Let me open a PR.